rocky/reports/views/report_overview.py

```diff
@@ -93,12 +96,119 @@ class ReportHistoryView(BreadcrumbsReportOverviewView, OctopoesView, ListView):
     paginator = RockyPaginator
     template_name = "report_overview/report_history.html"
 
+    def post(self, request, *args, **kwargs):
+        self.run_bulk_actions()
+        return self.get(request, *args, **kwargs)
+
     def get_queryset(self) -> ReportList:
         return ReportList(self.octopoes_api_connector, valid_time=self.observed_at)
 
+    def get_report_ooi(self, ooi_pk: str) -> Report:
+        return self.octopoes_api_connector.get(Reference.from_str(f"{ooi_pk}"), valid_time=datetime.now(timezone.utc))
+
+    def run_bulk_actions(self) -> None:
+        action = self.request.POST.get("action", "")
+        report_references = self.request.POST.getlist("report_reference", [])
+
+        if action == "rename":
+            return self.rename_reports(report_references)
+
+        if action == "delete":
+            return self.delete_reports(report_references)
+
+        if action == "rerun":
+            return self.rerun_reports(report_references)
+
+    def delete_reports(self, report_references: list[str]) -> None:
+        reports_selected = self.request.GET.get("report", "")
+        if reports_selected == "all":
+            all_reports = self.get_queryset()[:]  # must be sliced
```

Comment:
Why does it need to be sliced?
```suggestion
            all_reports = list(self.get_queryset())
```

Reply:
This got refactored in commit [92df2ba](https://github.com/minvws/nl-kat-coordination/pull/3777/commits/92df2bafec7f9c95e78e2e04288e531814018d37)